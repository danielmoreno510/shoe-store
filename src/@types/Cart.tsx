type CartProps = {
  cart: Shoe[];
  setCurrentShoe(shoe: Shoe): any;
  setClearCart(): any;
  navigation: Navigate;
};
