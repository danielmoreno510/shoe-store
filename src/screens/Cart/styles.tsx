import {StyleSheet} from 'react-native';

import {primaryColor} from '../../constants';
import {widthPercentageToDP} from '../../utils/responsiveUtils';

export default StyleSheet.create({
  container: {
    flex: 1,
    padding: widthPercentageToDP('5%'),
    backgroundColor: 'white',
  },
  shoes: {
    flexDirection: 'row',
    borderBottomColor: primaryColor,
    borderBottomWidth: 1,
    padding: 12,
  },
  empty: {
    fontSize: 24,
    marginTop: 20,
    textAlign: 'center',
  },
  buttonsContainer: {
    marginBottom: 40,
  },
  clearCar: {
    width: 140,
    alignSelf: 'center',
    margin: 20,
  },
});
