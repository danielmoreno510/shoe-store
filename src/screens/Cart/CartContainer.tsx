import Cart from './Cart';
import {StoreService} from '../../redux/StoreService';
import {putCurrentShoe, putClearCart} from '../../redux/actions';

export const mapStateToProps = state => ({
  cart: state.shoes.cart,
});

export const mapDispatchToProps = dispatch => ({
  setCurrentShoe: shoe => {
    dispatch(putCurrentShoe(shoe));
  },
  setClearCart: () => {
    dispatch(putClearCart());
  },
});

const CartContainer = StoreService.connect(mapStateToProps, mapDispatchToProps)(Cart);

export default CartContainer;
