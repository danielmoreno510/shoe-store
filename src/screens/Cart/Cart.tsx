import React from 'react';
import {ScrollView, View, Text} from 'react-native';

import Carousel from '../../components/common/Shoe/Carousel';
import ShoeInfo from '../../components/common/Shoe/ShoeInfo';
import Button from '../../components/common/Button';
import styles from './styles';

const Cart: React.FC<CartProps> = ({cart, setCurrentShoe, setClearCart, navigation}) => {
  const showProduct = (shoe: Shoe) => {
    navigation.navigate('Shoe Detail');
    setCurrentShoe(shoe);
  };

  return (
    <ScrollView style={styles.container}>
      {cart.length > 0 ? (
        cart.map((shoe, index) => (
          <View key={index} style={styles.shoes}>
            <Carousel shoe={shoe} width={100} />
            <ShoeInfo shoe={shoe} showProduct={() => showProduct(shoe)} oneSection newSize={shoe.size} />
          </View>
        ))
      ) : (
        <Text style={styles.empty}>You have no products in your cart</Text>
      )}
      {cart.length > 0 && (
        <View style={styles.buttonsContainer}>
          <View style={styles.clearCar}>
            <Button text="Clear Cart" onPress={() => setClearCart()} color="#af1313" />
          </View>
          <Button text="Pay" onPress={() => navigation.navigate('Pay')} />
        </View>
      )}
    </ScrollView>
  );
};

export default Cart;
