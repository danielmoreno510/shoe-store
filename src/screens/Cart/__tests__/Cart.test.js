import React from 'react';
import {shallow} from 'enzyme';

import Cart from '../Cart';

describe('Cart component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      cart: [],
      setCurrentShoe: jest.fn(),
      setClearCart: jest.fn(),
      navigation: {navigate: jest.fn()},
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<Cart {...testProps} />);

      it('THEN should display a regular Cart', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe('WHEN the showProduct and Clear Cart function is called', () => {
      const customProps = {...testProps, cart: [{}]};
      const shallowWrapper = shallow(<Cart {...customProps} />);
      shallowWrapper.find('ShoeInfo').props().showProduct();
      shallowWrapper.find('CustomButton').at(0).simulate('press');
      shallowWrapper.find('CustomButton').at(1).simulate('press');

      it('THEN setCurrentShoe and setClearCart function should be called', () => {
        expect(testProps.setCurrentShoe).toBeCalledTimes(1);
        expect(testProps.setClearCart).toBeCalledTimes(1);
      });
    });
  });
});
