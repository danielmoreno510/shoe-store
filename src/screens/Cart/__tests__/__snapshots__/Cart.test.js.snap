// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Cart 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 37.5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    You have no products in your cart
  </Text>
</ScrollView>
`;
